### PR TITLE
add API to collect_existing_environment_hooks()

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

File: colcon_core/task/__init__.py [hunk 1]
```diff
@@ -299,3 +301,25 @@ def install(args, rel_src, rel_dst):
...
+    :param Path path: The path to check for existing environment hooks
+    :rtype: list
+    """
+    additional_hooks = []
+    shell_extensions = get_shell_extensions()
+    file_extensions = OrderedDict()
+    for shell_extensions_same_prio in shell_extensions.values():
+        for shell_extension in shell_extensions_same_prio.values():
+            for file_extension in shell_extension.get_file_extensions():
+                file_extensions[file_extension] = shell_extension
```
Comment (jacobperron) [inline colcon_core/task/__init__.py:319]: `shell_extension` is not used  below. What about making `file_extensions` a set?

```suggestion
            file_extensions.update(shell_extension.get_file_extensions())
```

File: colcon_core/task/__init__.py [hunk 2]
```diff
@@ -299,3 +300,24 @@ def install(args, rel_src, rel_dst):
...
+    :param Path path: The path to check for existing environment hooks
+    :rtype: list
+    """
+    additional_hooks = []
+    shell_extensions = get_shell_extensions()
+    file_extensions = []
+    for shell_extensions_same_prio in shell_extensions.values():
+        for shell_extension in shell_extensions_same_prio.values():
+            file_extensions += shell_extension.get_file_extensions()
+    for file_extension in file_extensions:
```
Comment (jacobperron) [inline colcon_core/task/__init__.py:318]: Is it possible to have duplicate file extensions (ie. should we guard against this)?

Comment (dirk-thomas) [reply colcon_core/task/__init__.py:318]: No.

**Comment by dirk-thomas** [comment]
Closing in favor of the already existing functionality to specify `hooks` in a [colcon.pkg](https://colcon.readthedocs.io/en/released/user/configuration.html?highlight=hooks#colcon-pkg-files) file.

**Pull request closed without merging**